src/ezmsg/core/component.py

```diff
@@ -9,76 +9,98 @@
 
 from typing import List, Optional, Dict, Tuple, Any, Callable
 
+import logging
+
+logger = logging.getLogger("ezmsg")
+
 
 class ComponentMeta(ABCMeta):
     def __init__(
         cls, name: str, bases: Tuple[type, ...], fields: Dict[str, Any], **kwargs: Any
     ) -> None:
         super(ComponentMeta, cls).__init__(name, bases, fields)
 
-        cls.__settings_type__ = Settings
-        cls.__state_type__ = State
-
-        for base in bases:
-            if (
-                cls.__settings_type__ is not Settings
-                and hasattr(base, "__settings_type__")
-                and getattr(base, "__settings_type__") is not Settings
-            ):
-                # This means class settings has already been set by a base class.
-                raise Exception(
-                    f"Class {name} already has settings of type {cls.__settings_type__}, "
-                    + f"but is trying to be set to {getattr(base, '__settings_type__')}"
-                )
-            if (
-                hasattr(base, "__settings_type__")
-                and getattr(base, "__settings_type__") is not Settings
-            ):
-                cls.__settings_type__ = getattr(base, "__settings_type__")
-            if (
-                cls.__state_type__ is not State
-                and hasattr(base, "__state_type__")
-                and getattr(base, "__state_type__") is not State
-            ):
-                # This means class state has already been set by a base class.
-                raise Exception(
-                    f"Class {name} already has state of type {cls.__state_type__}, "
-                    + f"but is trying to be set to {getattr(base, '__state_type__')}"
-                )
-            if (
-                hasattr(base, "__state_type__")
-                and getattr(base, "__state_type__") is not State
-            ):
-                cls.__state_type__ = getattr(base, "__state_type__")
+        cls.__streams__ = {}
 
         if "SETTINGS" in cls.__annotations__:
-            cls.__settings_type__ = cls.__annotations__["SETTINGS"]
-
+            settings_cls = cls.__annotations__["SETTINGS"].__name__
+            logger.warning(
+                f"{cls.__name__} SETTINGS should be assigned rather than annotated: `SETTINGS = {settings_cls}` -> `SETTINGS = {settings_cls}`"
```

Comment:
f"{cls.__name__} SETTINGS should be assigned rather than annotated: `SETTINGS: {settings_cls}` -> `SETTINGS = {settings_cls}`"

Reply:
Updated to use `name` instead.

Reply:
Sorry, that was just GitHub messing up the formatting of my comment. I thnk `__name__` part is fine. I was commenting on the first part (to be changed) reading `SETTINGS =` while I think it should be `SETTINGS:` that you want to change.

Reply:
Ahh, I totally missed that, thanks.